implemented returning id in RESULTS_ARRAY

einorler opened the pull request:
closes #468

saimaz commented on Service/Manager.php:
```diff
@@ -589,15 +589,20 @@ private function parseResult($raw, $resultsType, $scrollDuration = null)
     {
-        if (array_key_exists('_source', $data)) {
-            return $data['_source'];
+        if (array_key_exists('_source', $data) && array_key_exists('_id', $data)) {
```
`_id` field always comes back in a response. No need to check for that. `_source` is checked in case of fields might be set in the query then `_source` won't come back.

saimaz commented on Service/Manager.php:
```diff
@@ -589,15 +589,20 @@ private function parseResult($raw, $resultsType, $scrollDuration = null)
-                $output[] = $item['_source'];
+                $output[$item['_id']] = [
+                    '_type'  => $item['_type'],
+                    '_index'  => $item['_index'],
```
Index will always be the same. Repository class works with 1 manager at the time.

saimaz commented on Service/Manager.php:
```diff
@@ -589,15 +589,20 @@ private function parseResult($raw, $resultsType, $scrollDuration = null)
             foreach ($data['hits']['hits'] as $item) {
-                $output[] = $item['_source'];
+                $output[$item['_id']] = [
+                    '_type'  => $item['_type'],
```
The same as in the comment below about index.

saimaz commented on Service/Manager.php:
```diff
@@ -589,15 +589,20 @@ private function parseResult($raw, $resultsType, $scrollDuration = null)
+                    '_index'  => $item['_index'],
+                    '_score' => $item['_score'],
+                ];
+                $output[$item['_id']] = array_merge($output[$item['_id']], $item['_source']);
```
Can we avoid array_merge here?

einorler commented:
Removed checking if `_id` is returned, removed the returning of `index` and `type` in the result array and changed the implementation of results formation to avoid `array_merge` function

saimaz commented on Service/Manager.php:
```diff
@@ -590,14 +590,15 @@ private function parseResult($raw, $resultsType, $scrollDuration = null)
             foreach ($data['hits']['hits'] as $item) {
-                $output[] = $item['_source'];
+                $output[$item['_id']] = $item['_source'];
+                $output[$item['_id']]['_score'] = $item['_score'];
```
`_score` is not needed at the array response

saimaz commented on Service/Manager.php:
```diff
@@ -590,14 +590,15 @@ private function parseResult($raw, $resultsType, $scrollDuration = null)
         if (isset($data['hits']['hits'][0]['_source'])) {
             foreach ($data['hits']['hits'] as $item) {
-                $output[] = $item['_source'];
+                $output[$item['_id']] = $item['_source'];
```
Do not change result structure. Append `_id` to the `$item` element.

einorler commented:
Removed `_source` from return array and changed back its structure, simply including an additional element with a key `_id`
